### PR TITLE
SL-1566: Fix the build process

### PR DESCRIPTION
**Pull request by XVincentX**
This PR puts some order in the build process of Prism. It was really confused and most likely wrong.

In particular:

1. It is impossible to install the dependencies at all because of an `.npmrc` file with a `NPM_TOKEN` env variable that I do not have. That file gets now injected by CircleCI at runtime so that, locally, I can leverage my user-level `.npmrc` file instead of having a customised one for this project.
2. The source code is not usable at all from VSCode. The references were wrong so it was giving me red warnings everywhere.
3. The build order and dependencies have not been defined correctly. This means that it was not possible to build the projects one by one; you had to build them all together. This would have make the single project releases almost impossible.
4. There was some test code relying on the library being compiled instead of requiring the source code directly, making the tests fail when I removed the build process from the test process.

This PR fixes all of these points by establishing the project dependencies and links so that:

* VSCode does not complain when opening the project.
* All the project can be built one by one.
* You can effectively test the project without having to compile it.
* You can navigate through the code and VSCode will recognise the internal references and point you to the right file
* Correctly make sure you're not relying on the compiled files

---

I do not believe this is the optimal configuration (yet), but at least this one works.

P.S: This PR is exactly the example of the *Typescript Tax*

### CODE REVIEW

**Comment by marbemac** [comment]
Nice work! @chris-miaskowski has done some work on this build stuff in the last few months - suggest he reviews given that experience.

**Comment by pytlesk4** [comment]
Should we close #173?

This is awesome though, good job man!

**Comment by XVincentX** [comment]
Given the nature of these changes and the intent of that PR — I'd be inclined to do so. I think there are also other changes that will need to be made here before making the release happen, and I can take care of these, but I guess these can be a next step.

File: packages/cli/tsconfig.json [hunk 1]
```diff
@@ -1,14 +1,7 @@
 {
-  "extends": "../../tsconfig.base.json",
+  "extends": "../tsconfig.json",
   "compilerOptions": {
-    "rootDir": "src",
-    "outDir": "lib",
-    "target": "es2017"
```
[the remark this answers is not in the document]
Comment (chris-miaskowski) [reply packages/cli/tsconfig.json]: Ok, I remember, I needed this when I was trying to add a unit test (but I didn't in the end). Probably safe to remove now but may cause some problems in the future if we want to add tests.

File: packages/cli/tsconfig.build.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    "rootDir": "src",
+    "outDir": "lib"
+  },
+  "include": ["src"],
+  "references": [
+    {
+      "path": "../http-server/tsconfig.build.json"
+    },
+    {
+      "path": "../http/tsconfig.build.json"
```
[the remark this answers is not in the document]
Comment (XVincentX) [reply packages/cli/tsconfig.build.json]: Correct! It's not needed!

**Comment by XVincentX** [comment]
@chris-miaskowski `tsc -b` uses the default file, which is the one needed for VSCode, not for the build.

The best would be to use `yarn build`, `yarn build.watch`, which will pre can the necessary parameters for you.

**Pull request merged**